### ISSUE TITLE
Consider using codecov.io

### ISSUE THREAD

**Issue by AlekSi** [umputun/remark42#51]
In my experience, it is much nicer than Coveralls.
See https://codecov.io/gh/AlekSi/remark and https://github.com/AlekSi/remark/pull/1.

**Comment by umputun** [comment]
I have considered codecov.io but they `bash <(curl -s https://codecov.io/bash)` kind of odd, to say the least. I looked over your repo and ran into a problem right away - it failed to show coverage of the source file - https://codecov.io/gh/AlekSi/remark/src/codecov/app/main.go. Probably can be configured but I'm not sure why it even need an extra configuration for the most basic functionality.  Coverage calculations don't make sense to me either.

**Comment by umputun** [comment]
Coveralls is down for the second day :( 

I have tried your way to integrate codecov.io and it seems to work fine. The only problem I have is a way we report to them. Loading bash script and running it in our ci feel very insecure to me.  Currently, we don't have sensitive secrets in CI, but one day we will, things like docker hub passwd and so on.

Is there any better/safer way to do it without running such potentially dangerous script?

**Comment by umputun** [comment]
tried to like it, but I just can't. Like coverall better

**Comment by AlekSi** [comment]
> I looked over your repo and ran into a problem right away - it failed to show coverage of the source file - https://codecov.io/gh/AlekSi/remark/src/codecov/app/main.go

It works for me: 
![2018-05-27_13-27-18](https://user-images.githubusercontent.com/11512/40584881-b7518484-61b1-11e8-82cf-89e4205123c5.png)

> Loading bash script and running it in our ci feel very unsecure to me.

I remember reading about their motivation somewhere, but I can't find it now. Basically, they were saying that using a downloaded bash script is as secure as using any other installed software. You trust alpine packagers when installing curl, mattn when installing goveralls with `go get`, etc. `go get` is very close to `curl+bash` in a sense that both do not have any signatures.
You can also download a current version of the script, review it, and then add it to the base image.

> Is there any better/safer way to do it without running such potentially dangerous script?

Well, there is an API: https://docs.codecov.io/v4.3.0/reference#upload But I don't know if anyone re-implemented an own uploader. It should not be hard though.

**Comment by umputun** [comment]
> It works for me

At the time I have tried this main.go link it showed a bizarre message asking to set up access to a private repo.

> go get is very close to curl+bash in a sense that both do not have any signatures.

Good point. I'll add some vendoring into build-go image to prevent such downloads and work with blessed packages only.

> you can also download a current version of the script, review it, and then add it to the base image.

Yeah, I've tried to do it. But this thing is enormous (for sh script), and such level of bash magic is not something I can reliably verify.

The bottom line - this bash thingy feels wrong to me. Besides, I don't see an obvious reward by switching as coveralls does the job and provides useful reports. I would argue - more convenient, but this is the matter of taste and expectations.
